src/gui/qgsbrowserdockwidget.h

```diff
@@ -62,94 +84,169 @@ class QgsBrowserPropertiesWidget : public QWidget
+ */
+class GUI_EXPORT QgsBrowserPropertiesDialog : public QDialog, private Ui::QgsBrowserPropertiesDialogBase
 {
     Q_OBJECT
   public:
+
+    /**
+      * Constructor for QgsBrowserPropertiesDialog
+      * \param settingsSection prefix for settings (from the object name)
+      * \param parent parent widget
+      */
     QgsBrowserPropertiesDialog( const QString &settingsSection, QWidget *parent = nullptr );
     ~QgsBrowserPropertiesDialog();
 
+    //! Create dialog from the given item and add it
     void setItem( QgsDataItem *item );
 
   private:
     QgsBrowserPropertiesWidget *mPropertiesWidget = nullptr;
     QString mSettingsSection;
 };
 
+/**
+ * \ingroup gui
+ * The QgsBrowserDockWidget class
+ * \since QGIS 2.10
+ */
 class GUI_EXPORT QgsBrowserDockWidget : public QgsDockWidget, private Ui::QgsBrowserDockWidgetBase
 {
     Q_OBJECT
   public:
+
+    /**
+      * Constructor for QgsBrowserDockWidget
+      * \param name name of the widget
+      * \param parent parent widget
+      */
     explicit QgsBrowserDockWidget( const QString &name, QWidget *parent = nullptr );
     ~QgsBrowserDockWidget();
+    //! Add directory to favorites
     void addFavoriteDirectory( const QString &favDir );
 
   public slots:
+    //! Add layer at index
     void addLayerAtIndex( const QModelIndex &index );
+    //! Show context menu
     void showContextMenu( QPoint );
 
+    //! Add current item to favorite
     void addFavorite();
+    //! Add directory from file dialog to favorite
     void addFavoriteDirectory();
+    //! Remove from favorite
     void removeFavorite();
 
+    //! Refresh browser view model (and view)
     void refresh();
 
+    //! Show/hide filter widget
     void showFilterWidget( bool visible );
+    //! Enable/disable properties widget
     void enablePropertiesWidget( bool enable );
+    //! Set filter syntax
     void setFilterSyntax( QAction * );
+    //! Set filter case sensitivity
     void setCaseSensitive( bool caseSensitive );
+    //! Apply filter to the model
     void setFilter();
+    //! Update project home directory
     void updateProjectHome();
 
 
-    // layer menu items
+    //! Add selected layers to the project
     void addSelectedLayers();
+    //! Show the layer properties
     void showProperties();
+    //! Hide current item
     void hideItem();
+    //! Toggle fast scan
     void toggleFastScan();
 
+    //! Selection hass changed
     void selectionChanged( const QItemSelection &selected, const QItemSelection &deselected );
+    //! Splitter has been moved
     void splitterMoved();
 
   signals:
-    // A file needs to be opened
-    void openFile( const QString );
-    // Drop uri list needs to be handled
-    void handleDropUriList( QgsMimeDataUtils::UriList );
+    //! Emitted when a file needs to be opened
+    void openFile( const QString & );
+    //! Emitted when drop uri list needs to be handled
+    void handleDropUriList( const QgsMimeDataUtils::UriList & );
 
   protected:
```

Comment:
I think this whole section can be made private instead (apart from showEvent() which is protected from QWidget) - those methods were not really meant for re-use in further subclasses. Turning the methods to private will remove them from python interface altogether and will make the class API simpler.